src/api/models/lines.js

```diff
@@ -43,49 +45,55 @@ lineSchema.plugin(idvalidator);
 
 // create a LineEvent entry to keep track of history.
 // should be called after every update event.
-lineSchema.methods.logEvent = function() {
+// returns a promise (resolves to the line that logEvent() is called on)
+lineSchema.methods.logEvent = async function() {
     console.log("Line: Logging line update event");
     var event = LineEvent();
     event.user_id = this.user_id;
     event.employer_id = this.employer_id;
     event.status = this.status;
-    event.save(function(err) {
-        if (err)
-            return console.log("LineLogError: Unable to log line event: " + err);
-    });
+    try {
+        await event.save();
+        return this;
+    } catch (err) {
+        console.log("LineLogError: Unable to log line event: " + err);
+        return Promise.reject("LineLogError: Unable to log line event: " + err);
+    }
 }
+
 // update status of line
 // used for manually updating status, instead of using api route
 // also creates log event.
-// returns 'success' upon success, and an error message otherwise.
+// returns a promise: 'success' upon success, and an error message otherwise.
 lineSchema.methods.updateStatus = async function(status) {
     var msg = null;
     //verify that line progression is happening in order
+    //console.log('attempting to update line from %s to %s', this.status, status); //DEBUG
     switch (status) {
 
         case 'notification':
             if (this.status !== 'preline') {
-                msg = "LineUpdateError: Cannot move to notification status from status " + this.status;
+                msg = "LineUpdateError: Cannot move to notification status from status " + status;
             }
             break;
         case 'inline':
-            if (this.status !== 'notification') {
-                msg = "LineUpdateError: Cannot move to inline status from status " + this.status;
+            if (this.status !== 'notification' && this.status !== 'preline') {
```

Comment:
so can u think of a case where a preliner can illegally join inline even if BATCH is full? if not this is fine.

Reply:
if they figure out our API then they can

Reply:
i'll make an issue for this

Reply:
#114